### PR TITLE
Use librsvg for PNG rendering

### PR DESCRIPTION
**Pull request by physikerwelt**
* Remove batik support
* Use librsvg
* Breaking change: Requires additional package
  librsvg2-dev see https://www.npmjs.com/package/librsvg

### CODE REVIEW

**Comment by physikerwelt** [comment]
@wikimedia/math this is a really huge performance benefit, and no temp files for png images anymore

**Comment by physikerwelt** [comment]
cc @pkra

**Comment by pkra** [comment]
Thanks for the cc. I'm guessing that means you want to submit this upstream at mathjax-node as well?  The first question that comes to mind: could you abstract the converter? It'd be great if people could switch out their personal favorite, in particular svg2png is a fan favorite.

**Comment by pkra** [comment]
But it would be good to make a PR on mathjax-node to discuss this further.

Oh and also: really great to see this! :100:

**Comment by physikerwelt** [comment]
@pkra I would love to make that this modular. I was thinking about that, but I'm not not sure what's the best way to do this. Maybe @d00rman has an idea.
With this change the performance improves drastically
https://github.com/physikerwelt/mathoid-server/commit/5169289574b91fcd6c93eebaf2b54cfa96919084
However, there is additional effort required to make this run on windows. Took me about 1 hour, but only since I called by directory "C:\GTK" instad of "C:\gtk" which is the same for windows but not for node.

**Comment by pkra** [comment]
>  I was thinking about that, but I'm not not sure what's the best way to do this. 

Maybe discuss upstream?

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,13 @@
 -sudo: false
 language: node_js
 node_js:
-  - "4.2.4"
+  - "4.2"
```
Comment (d00rman) [inline .travis.yml]: We want to keep `4.2.4` because that's the version used in WMF prod.

**Comment by d00rman** [comment]
In principle, I welcome this change, but I'm afraid WMF isn't ready for it. We have our own pkgs for `librsvg` which don't currently exist for Jessie :/ We need to follow up on this, though.

File: .travis.yml (hunk 1 above)
Comment (physikerwelt) [inline .travis.yml]: ok. But we should keep that consistent. I think in other repos we have just 4.2

**Comment by physikerwelt** [comment]
@d00rman did you file a bug for librsvg-dev? I think the absence of PNG export is the most important blocker for further developments in the math environment. We might consider taking a special path for math since we do not feed arbitrary svg images to librsvg.

**Comment by pkra** [comment]
Give me a day to finish up some work for an upstream modularization of the png generation.

**Comment by d00rman** [comment]
@physikerwelt I've just spoken to @Stype about it, and he says that the newest Jessie librsvg contain all of our security fixes, so we are actually good to go on that front. `libsrvg2-dev` is present in Jessie.

File: .travis.yml (hunk 1 above)
Comment (d00rman) [inline .travis.yml]: We'll soon switch to node 4.3, so you might as well put that instead :)

**Comment by d00rman** [comment]
@physikerwelt module the minor Travis config nits, LGTM, so when these are corrected I'd be happy to merge.

File: .travis.yml (hunk 1 above)
Comment (gwicke) [inline .travis.yml]: You could specify just "4", which pulls in the latest 4.x automatically.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mathoid-mathjax-node",
-  "version": "0.5.0",
+  "version": "0.5.1",
```
Comment (d00rman) [inline package.json]: Actually, since this a breaking change, v0.6.0 would be more appropriate.

File: lib/mj-single.js [hunk 3]
```diff
@@ -501,33 +505,38 @@ function GetSVG(result) {
...
-          result.png = "data:image/png;base64,"+(buffer||"").toString('base64');
-          fs.unlinkSync(tmpSVG);
-          fs.unlinkSync(tmpPNG);
-          check(err); synch();
-        });
-      });
-    });
-    return synch;  // This keeps the queue from continuing until the readFile() is complete
-  }
+    var Readable = require('stream').Readable;
```
Comment (d00rman) [inline lib/mj-single.js]: Any reason why this is not in the file's header?

Comment (physikerwelt) [inline lib/mj-single.js]: oh no not really.Should I move it?

Comment (d00rman) [inline lib/mj-single.js]: Yup, please

Comment (physikerwelt) [inline lib/mj-single.js]: @d00rman done

File: lib/mj-single.js [hunk 4]
```diff
@@ -501,33 +506,37 @@ function GetSVG(result) {
...
-      });
-    });
-    return synch;  // This keeps the queue from continuing until the readFile() is complete
-  }
+    var s = new Readable();
+    var pngScale = 2;
+    var svgFile = result.svgfile; delete result.svgfile;
+    var width = result.width; delete result.width;
+    var height = result.height; delete result.height;
+    if (data.png) {
```
Comment (d00rman) [inline lib/mj-single.js]: If `data.png` is false-y, no return is specified. Shouldn't `synch` be retuned anyway?

File: lib/mj-single.js [hunk 5]
```diff
@@ -501,33 +506,37 @@ function GetSVG(result) {
...
-          fs.unlinkSync(tmpPNG);
-          check(err); synch();
-        });
-      });
-    });
-    return synch;  // This keeps the queue from continuing until the readFile() is complete
-  }
+    var s = new Readable();
+    var pngScale = 2;
+    var svgFile = result.svgfile; delete result.svgfile;
```
Comment (d00rman) [inline lib/mj-single.js]: It's much faster to use `result.svgfile = undefined` than `delete`.

File: lib/mj-single.js (hunk 4 above)
Comment (physikerwelt) [inline lib/mj-single.js]: I did not change that...

File: lib/mj-single.js (hunk 5 above)
Comment (physikerwelt) [inline lib/mj-single.js]: done...also for the next two lines

**Pull request merged**